code/modules/mob/living/basic/illusion/illusion.dm

```diff
@@ -0,0 +1,88 @@
+/mob/living/basic/illusion
+	name = "illusion"
+	desc = "It's a fake!"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "static"
+	icon_living = "static"
+	icon_dead = "null"
+	gender = NEUTER
+	mob_biotypes = NONE
+	melee_damage_lower = 5
+	melee_damage_upper = 5
+	combat_mode = TRUE
+	attack_verb_continuous = "gores"
+	attack_verb_simple = "gore"
+	maxHealth = 100
+	health = 100
+	speed = 0
+	faction = list(FACTION_ILLUSION)
+	basic_mob_flags = DEL_ON_DEATH
+	death_message = "vanishes into thin air! It was a fake!"
+	ai_controller = /datum/ai_controller/basic_controller/illusion
+	/// Weakref to what we're copying
+	var/datum/weakref/parent_mob_ref
+	/// Prob of getting a clone on attack
+	var/multiply_chance = 0
+	/// The blackboard key we want to set for our target
+	var/target_key = BB_BASIC_MOB_CURRENT_TARGET
+
+/mob/living/basic/illusion/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_HOSTILE_PRE_ATTACKINGTARGET, PROC_REF(attack_override))
+	RegisterSignal(src, COMSIG_LIVING_HEALTH_UPDATE, PROC_REF(on_health_changed))
```

Comment:
`COMSIG_LIVING_HEALTH_UPDATE` isn't an appropriate place to react to damage taken, because it's 1. triggered on healing and 2. does not even imply health changed

Use `COMSIG_MOB_AFTER_APPLY_DAMAGE`